ftdetect/pug.vim

```diff
@@ -0,0 +1 @@
+au BufRead,BufNewFile *.pug		set filetype=pug
```

Comment:
I'd rather do
```suggestion
au BufRead,BufNewFile *.pug		setlocal filetype=pug
```

Reply:
Fine by me, I've just peeked into the existing ftdetect rules [ftdetect/gomod.vim](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/ftdetect/gomod.vim) and [ftdetect/gmd.vim](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/ftdetect/gdscript.vim) are doing the same with ``set`` instead of ``setlocal``. Maybe they should be changed too?

Reply:
changed it to setlocal, which should make much more sense for most of these autocomands. Like i said, maybe worth changing the other 2 ftdetect scripts the same way.

Reply:
I agree on changing those we'll do it in another pr!